modules/snigelBidAdapter.js

```diff
@@ -1,19 +1,25 @@
 import {config} from '../src/config.js';
 import {registerBidder} from '../src/adapters/bidderFactory.js';
 import {BANNER} from '../src/mediaTypes.js';
-import {deepAccess, isArray, isFn, isPlainObject, inIframe, getDNT} from '../src/utils.js';
+import {deepAccess, isArray, isFn, isPlainObject, inIframe, getDNT, generateUUID} from '../src/utils.js';
 import {hasPurpose1Consent} from '../src/utils/gpdr.js';
 import {getGlobal} from '../src/prebidGlobal.js';
+import {getStorageManager} from '../src/storageManager.js';
 
 const BIDDER_CODE = 'snigel';
 const GVLID = 1076;
 const DEFAULT_URL = 'https://adserv.snigelweb.com/bp/v1/prebid';
 const DEFAULT_TTL = 60;
 const DEFAULT_CURRENCIES = ['USD'];
 const FLOOR_MATCH_ALL_SIZES = '*';
+const SESSION_ID_KEY = '_sn_session_pba';
 
 const getConfig = config.getConfig;
+const storageManager = getStorageManager({bidderCode: BIDDER_CODE});
 const refreshes = {};
+const pageViewId = generateUUID();
+const pageViewStart = new Date().getTime();
```

Comment:
This will not work in single page environments, I guess. You may want to overthink this implementation.